department-of-veterans-affairs/caseflow
# J hoang/appeals 34906

Resolves [34906](https://jira.devops.va.gov/browse/APPEALS-34906)

# Description
Upload all Correspondence Documents to eFolder if the general user clicks the "Intake appeal" button on the Correspondence Review Package page. If the upload fails, 

1. A error banner will display notifying the user that upload has failed and they may try again later. 
2. A EfolderUploadFailedTask will be created as a child of the ReviewPackageTask and given a status of in_progress.
3. The user will not be directed to the 10182 intake appeal process

## Acceptance Criteria

## Testing Plan

1. Login as MAIL_TEAM_SUPERVISOR_MAIL_INTAKE_USER
2. Click on "Your Queue"
3. Click on "Switch Views"
4. Click on "Correspondence Cases"
5. Click on a Correspondence case that has a package document type of 10182 (first 10 cases)
6. Click on "Intake appeal" button
7a. If upload failed, an error banner should appear and user should not be directed to intake.
7b. If upload was successful, user should be directed to intake.

# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
